docs/closure-compiler.md

```diff
@@ -1,179 +0,0 @@
-<!--docs:
-title: "Closure Compiler Annotations"
-layout: landing
-section: docs
-path: /docs/closure-compiler/
--->
-
-# MDC Web Authoring Practices for Interoperability with Closure Compiler
-
-## Who this document is for
-
-This document is for _core contributors to MDC Web, as well as contributors who wish to author
-new components, or make non-trivial changes to existing components._ It assumes you're familiar with
-our codebase, and have read through most of our [Authoring Components guide](./authoring-components.md).
-
-## Why this is needed
-
-MDC Web - and Material Design in general - was created by Google. Therefore, it is not only a top
-priority that MDC Web works seamlessly for our external community, but also that _MDC Web works
-seamlessly for all Google applications_.
-
-At Google, all JavaScript is processed and minified by the
-[Closure Compiler](https://github.com/google/closure-compiler) (which will be referred to as
-**closure**, the **compiler**, or any combination of those terms). MDC Web is now in TypeScript, but will still need to
-be transpiled to closure-annotated ES2015 for use at Google.
-
-## MDC Web Type System
-
-The overall type system is relatively straightforward, and boils down to 3 main concepts:
-
-- **Adapters are simply interfaces with a predefined shape specifying functions.** There is no such thing as a
-  "base adapter".
-- **Foundations are parameterized by their adapters.** For example, an `MDCRippleFoundation` would be
-  parameterized by an `MDCRippleAdapter`. Thus, the `MDCRippleFoundation` class is declared as
-  `class MDCRippleFoundation extends MDCFoundation<MDCRippleAdapter>`.
-- **Components are parameterized by their foundations.** For example, an `MDCRipple` would be
-  parameterized by `MDCRippleFoundation`, which - as shown above - is itself parameterized by
-  `MDCRippleAdapter`. Thus, the `MDCRipple` class is declared as
-  `class MDCRipple extends MDCComponent<MDCRippleFoundation>`.
-
-## MDC Web Closure Conventions
-
-The following guidelines outline the general conventions for writing closure-friendly code for MDC Web.
-This section should contain most - if not all - of what you need to get up and running writing components in our codebase.
-
-### All `import` statements must _not_ use re-exported modules.
```

Comment:
This should effectively be moved to best_practices.md in the form of a heading explaining that in the context of component packages, only the index and component files are allowed to reference `index`, because wrappers shouldn't need to pull in `index` or `component`

Reply:
Is this line still accurate? 

This is an unfortunate side-effect of how [closure's module naming mechanism works](https://github.com/google/closure-compiler/issues/2257).